UIKit Builder pattern.md

```diff
@@ -0,0 +1,172 @@
+---
+
+tags: iOS, MacOS, Swift
+
+author: Phan Viet Trung
+
+---
+## UIKit builder pattern.
+
+With the UIKit, to write a simple Login form we're usually do:
+```
+let txtUserName = UITextField()
+txtUserName.placeHolder = "User Name"
+txtUserName.textColor = .black8
+//For text change event.
+txtUserName.delegate = self 
+
+let txtPassword = UITextField()
+txtPassword.placeHolder = "Password"
+txtPassword.textColor = .black8
+//For hide password
+txtPassword.stype = .password
+txtPassword.delegate = self
+```
+
+
+For the small project, it totally fine. However with large project, we have a standard design. How can we reuse our code, keep it simple to use, and keep a flexility.
+
+Normally we will override the Original data type and create our own components following the style that designer give us.
+
+ex:
+```
```

Comment:
Is that `shift` language? Consider adding a language flag for a better readability with syntax highlight:
```suggestion
```shift
```